OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs

```diff
@@ -1,60 +1,140 @@
-﻿using UnityEngine;
+﻿using System.IO;
+using System;
+using UnityEngine;
 using UnityEditor;
 using UnityEditor.Callbacks;
-using System.Collections;
-using System.IO;
+using UnityEditor.iOS.Xcode;
+using UnityEditor.iOS.Xcode.Extensions;
+using System.Text;
+
+/*
+   Adds required frameworks to the iOS project, and adds the OneSignalNotificationServiceExtension
+   Also handles making sure both targets (app and extension service) have the correct dependencies
+*/
 
 #if UNITY_IPHONE && UNITY_EDITOR
 
-public struct framework {
-   public string name;
-   public string id;
-   public string fileId;
+public class BuildPostProcessor
+{
+   [PostProcessBuildAttribute(1)]
+   public static void OnPostProcessBuild(BuildTarget target, string path)
+   {
+      if (target == BuildTarget.iOS)
```

Comment:
Can remove this this target if, `#if UNITY_IPHONE && UNITY_EDITOR` already accounts for this.